### PR TITLE
fix: Watermark close fix and removed the nil check

### PR DESCRIPTION
**Pull request by jy4096**
Fixes: #224
Fixes: #230

### CODE REVIEW

File: pkg/watermark/generic/jetstream/generic.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func BuildWatermarkProgressors(ctx context.Context, vertexInstance *v1alpha1.Ver
 
 	var fetchWatermark fetch.Fetcher
 	if fromBuffer.Type == v1alpha1.SourceBuffer {
+		// TODO: does it use a non-nil hbWatch?
```
Comment (chromevoid) [inline pkg/watermark/generic/jetstream/generic.go:53]: Meaning...
is this really executed for source vertex?
https://github.com/numaproj/numaflow/blob/e953699e321baae2c46af9e7b2bce56b5ba935ce/pkg/watermark/fetch/processor_manager.go#L55-L57

Comment (vigith) [reply pkg/watermark/generic/jetstream/generic.go:53]: also #230

Comment (vigith) [reply pkg/watermark/generic/jetstream/generic.go:53]: can you check the source's code and see whether we need this check?

Comment (chromevoid) [reply pkg/watermark/generic/jetstream/generic.go:53]: 👀

**Review by whynowy — APPROVED** [review]
LGTM

**Review by vigith — APPROVED** [review]

**Pull request merged**